svg-2-android-vector/src/main/java/com/quittle/svg2androidvector/Svg2AndroidVectorPlugin.java

```diff
@@ -53,14 +56,25 @@ public void apply(final Project project) {
     }
 
     private static void onAfterEvaluate(final Project project, final Svg2AndroidVectorExtension extension) {
+        // The base Android extension, where all resources can be found.
+        final BaseExtension androidExtension = project.getExtensions().findByType(BaseExtension.class);
+        if (androidExtension == null) {
+            // This might mean it was applied to a non-Android project, possibly just the root project.
+            return;
+        }
         final TaskContainer taskContainer = project.getTasks();
         final Task parentTask = taskContainer.create(CONVERSION_PARENT_TASK_NAME);
         // An early Android task all the conversion tasks should be a dependency of
-        taskContainer.findByName(EARLY_ANDROID_TASK_NAME).dependsOn(parentTask);
+        final AppExtension androidApp = project.getExtensions().findByType(AppExtension.class);
+        final LibraryExtension androidLib = project.getExtensions().findByType(LibraryExtension.class);
+        final PreBuildDependencyAdder dependencyAdder = new PreBuildDependencyAdder(parentTask);
+        if (androidApp != null) {
+            androidApp.getApplicationVariants().all(dependencyAdder);
+        } else if (androidLib != null) {
+            androidLib.getLibraryVariants().all(dependencyAdder);
+        }
```

Comment:
Can you also handle the `else` case? Should it just be a warning or would it make sense to throw if there was no variant to act on.

Reply:
well, it would mean e.g. that I missed an option like TestExtension :)